recipe/patches/fix_arrow_libtool.patch

```diff
@@ -0,0 +1,15 @@
+diff --git a/cmake/arrow.txt.in b/cmake/arrow.txt.in
+index 5b3105102..9c40503ad 100644
+--- a/cmake/arrow.txt.in
++++ b/cmake/arrow.txt.in
+@@ -9,8 +9,8 @@ project(arrow-download NONE)
+
+ include(ExternalProject)
+ ExternalProject_Add(apachearrow
+-  GIT_REPOSITORY    https://github.com/apache/arrow.git
+-  GIT_TAG           apache-arrow-17.0.0
++  GIT_REPOSITORY    https://github.com/ProspectiveCo/arrow.git
++  GIT_TAG           tom-libtool-fix2
```

Comment:
No

Reply:
That tag includes [a patch](https://github.com/apache/arrow/pull/44385/commits/f2b12a311d1a14ab5d9f67428201e1389748a9c1) to work around an [obscure issue](https://github.com/apache/arrow/issues/44384) in Arrow when building it with bundled dependencies when /usr/bin/libtool is broken (as it seems to be on the conda-forge cross-compiling Mac builders). 

The bespoke tag on the Prospective fork was an expedient way to test that change turned builds green. Aside from removing the Mac builders and waiting for an arrow release which includes https://github.com/apache/arrow/pull/44385/, the alternative is to patch arrow.txt.in to add a `PATCH_COMMAND` which includes the arrow patch. I'll do that.

Reply:
Patch isn't sufficient, need to make perspective's build read `CMAKE_ARGS` from the environment too. In the python build, cmake is invoked from perspective-server's build script using the `cmake` crate.  This is where it's rocky: the cmake crate doesn't support passing arbitrary arguments to cmake. As of 2017 they said they'd accept a PR to add the feature. [open issue](https://github.com/rust-lang/cmake-rs/issues/27)

Two ways occur to me solve this:
- parse CMAKE_ARGS in the Rust build script from the environment and turn each parsed arg into a method call on the cmake::Config builder object, e.g. `cmake.define(var, var)` which would define
- parse CMAKE_ARGS from the environment in CMakeLists.txt and turn them into`set(${var} ${value})` and other cmake calls

could also side-step this and just patch the the cmake invocation in FindInstallDependency.cmake to pass on `-DCMAKE_LIBTOOL=$ENV{LIBTOOL}` but I would rather make perspective's build respect all CMAKE_ARGS if it isn't too much work